components/Ripple/src/MDCRippleTouchController.m

```diff
@@ -15,7 +15,28 @@
 #import "MDCRippleTouchController.h"
 
 @implementation MDCRippleTouchController {
+  MDCRippleView *_rippleView;
   BOOL _tapWentOutsideOfBounds;
+  BOOL _deferred;
+
+  struct {
+    unsigned int rippleTouchControllerShouldProcessRippleTouchesAtTouchLocation : 1;
+    unsigned int rippleTouchControllerDidProcessRippleViewAtTouchLocation : 1;
+    unsigned int rippleTouchControllerInsertRippleViewIntoView : 1;
+  } _delegateFlags;
+}
+
+- (nonnull instancetype)initWithView:(nonnull UIView *)view deferred:(BOOL)deferred {
+  self = [self init];
+  if (self) {
+    _deferred = deferred;
+    if (deferred) {
+      [self attachGestureRecognizerToView:view];
+    } else {
+      [self configureRippleWithView:view];
+    }
+  }
+  return self;
 }
 
 - (instancetype)initWithView:(UIView *)view {
```

Comment:
Tip: Please have `initWithView:` call `initWithView:deferred:` instead of keeping the implementations independent.